src/rules/property-unit-whitelist/index.js

```diff
@@ -0,0 +1,47 @@
+import { isObject } from "lodash"
+import valueParser from "postcss-value-parser"
+
+import {
+  declarationValueIndexOffset,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "property-unit-whitelist"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (p, u) => `Unexpected unit "${u}" for property "${p}"`,
+})
+
+export default function (whitelist) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: whitelist,
+      possible: [isObject],
+    })
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+
+      const { prop, value } = decl
+      const propWhitelist = whitelist[prop]
+
+      if (!propWhitelist) { return }
+
+      valueParser(value).walk(function (node) {
+        const unit = valueParser.unit(node.value).unit
+
+        if (unit && propWhitelist.indexOf(unit) === -1 && node.type !== "string") {
```

Comment:
Sorry but I thought of one other easy optimization here. It probably applies to the other whitelist/blacklist rules, as well, if you feel like retrofitting ...

So the `node.type` check can go before the use of `valueParser`, thereby saving the parsing step if the node is a string. 

```js
valueParser(value).walk(function (node) {
  if (node.type === "string") { return }
  const unit = valueParser.unit(node.value).unit
```